src/model/events.js

```diff
@@ -6,19 +6,60 @@ export default class Events extends Observer {
     this._events = [];
   }
 
-  setEvents(events) {
+  setEvents(updateType, events) {
     this._events = events.slice();
+    this._notify(updateType);
   }
 
   getEvents() {
     return this._events;
   }
 
+  static adaptToClient(event) {
+    const adaptedEvent = Object.assign(
+        {},
+        event,
+        {
+          price: event.base_price,
+          isFavorite: event.is_favorite,
+          dateFrom: event.date_from,
+          dateTo: event.date_to
+        }
+    );
+
+    delete adaptedEvent.is_favorite;
+    delete adaptedEvent.base_price;
+    delete adaptedEvent.date_from;
+    delete adaptedEvent.date_to;
+
+    return adaptedEvent;
+  }
+
+  static adaptToServer(event) {
+    const adaptedEvent = Object.assign(
+        {},
+        event,
+        {
+          "base_price": event.price,
+          "is_favorite": event.isFavorite,
+          "date_from": event.dateFrom,
+          "date_to": event.dateTo
+        }
+    );
+
+    delete adaptedEvent.isFavorite;
+    delete adaptedEvent.price;
+    delete adaptedEvent.dateFrom;
+    delete adaptedEvent.dateTo;
+
+    return adaptedEvent;
```

Comment:
Здесь примерно так-же:
```suggestion
    return {
      id: event.id,
      destination: event.destination,
      offers: event.offers,
      type: event.type,
      [`base_price`]: event.price,
      [`is_favorite`]: event.isFavorite,
      [`date_from`]: event.dateFrom,
      [`date_to`]: event.dateTo,
    };
```
Плюс, можно избавиться от дублирования общих полей, создав хелпер `getCommonFields`, который бы возвращал объект с `id, type, offers, destination`

Reply:
поправил